api/provider/aws/service_create.go

```diff
@@ -148,31 +148,17 @@ func (s *ServiceProvider) Create(req models.CreateServiceRequest) (string, error
 		}
 	}
 
-	fn := func() (shouldRetry bool, err error) {
-		if err := s.createService(
-			cluster,
-			serviceName,
-			taskDefinitionARN,
-			loadBalancerRole,
-			networkMode,
-			req.Stateful,
-			scale,
-			subnets,
-			securityGroupIDs,
-			loadBalancer,
-		); err != nil {
-			if strings.Contains(err.Error(), "Unable to assume role") {
-				log.Printf("[DEBUG] Failed service create, will retry (%v)", err)
-				return true, nil
-			}
-
-			return false, err
-		}
-
-		return false, nil
-	}
-
-	if err := retry.Retry(fn, retry.WithTimeout(time.Second*30), retry.WithDelay(time.Second)); err != nil {
+	if err := s.waitUntilServiceCreated(
+		cluster,
+		serviceName,
+		taskDefinitionARN,
+		loadBalancerRole,
+		networkMode,
+		req.Stateful,
+		scale,
+		subnets,
+		securityGroupIDs,
+		loadBalancer); err != nil {
 		if err, ok := err.(awserr.Error); ok && err.Code() == "InvalidParameterException" {
```

Comment:
I know this isn't something you wrote - but what's the point of this line? Seems unnecessary since we don't do anything when we catch that error code.